laiaaa0/assignment2/BinaryTree.cpp

```diff
@@ -112,53 +95,51 @@ int BinaryTree<T>::Depth(){
 
 
 template<class T>
-const T BinaryTree<T>::LowestCommonAncestor (const T & key1, const T & key2){
-  T lowest_common_ancestor;
+bool BinaryTree<T>::LowestCommonAncestor (const T & key1, const T & key2,  T & lca){
 
-  BinaryTree<T> *tree_of_key_1;
-  BinaryTree<T> *tree_of_key_2;
-
-  bool key1_in_tree = this->IsValueInTree(key1,tree_of_key_1);
-  bool key2_in_tree = this->IsValueInTree(key2,tree_of_key_2);
-
-  if (!key1_in_tree or !key2_in_tree){
-    throw std::invalid_argument("key not in tree!");
+  if (this == nullptr){
+    return false;
   }
   else {
-    std::vector<T> path_from_key1_to_root;
+    BinaryTree<T> *tree_of_key;
 
-    BinaryTree<T> * trees_in_path_key1 = tree_of_key_1;
+    bool foundInLeft1 = this->left_subtree_->FindValueInTree(key1,tree_of_key);
```

Comment:
this->left_subtree_ can be nullptr, right?
If you find it more natural you can create a private (maybe static) function FindValueInTree that takes a node * to the subtree you are interested on, this parameter could be nullptr.